content/en/blog/news/Enhancing_Performance_in_Microservice_Architecture_with_Kitex/Enhancing_Performance_in_Microservice_Architecture_with_Kitex.md

```diff
@@ -0,0 +1,538 @@
+
+## Is Faster Downstream Parallel Processing Always Better? - Concurrency vs. Timeout
+
+Knowing the upper bound for physical parallel computing in a program is considerably high, we can leverage this insight to increase or decrease the number of working threads (GOMAXPROCS) or adjust the degree of concurrency within the program.
+
+Let's consider a calling scenario where the business sends requests to the same upstream with four concurrent processes. Each request upstream requires 50ms of processing time. Based on this, the downstream sets the timeout time to **100ms**. 
+
+Though this seems reasonable, if, at that time, the upstream had only two CPUs available to handle requests (which also have to manage other work or perform Garbage Collection activities), the third RPC request would likely time out.
+
+![Image](/img/blog/Enhancing_Performance_in_Microservice_Architecture_with_Kitex/29.jpeg)
+
+However, simply reducing concurrency isn't always the solution and it doesn't benefit all cases. 
+
+![Image](/img/blog/Enhancing_Performance_in_Microservice_Architecture_with_Kitex/30.jpeg)
+
+If upstream's computational resources are abundantly available, increasing concurrency could efficiently utilize the processing power upstream.
+
+## Balancing Resource Utilization – Reserving Computational Capacity for Other Processes
+
+If there are multiple processes within the container, it's essential to reserve resources for these operations. This consideration is particularly crucial in scenarios like the deployment of a Service Mesh data plane, where the same container operates as a sidecar. 
+
+If a downstream process utilizes the entirety of the time slice allocated within a computation cycle, it's highly probable that it'll face resource throttling when it's the upstream process's turn, which could subsequently affect the service’s latency.
+
+## Optimizing Service Concurrency Degree
+
+* **Adjust the Number of Work Threads**: For instance, the `GOMAXPROCS` directive in Go allows us to modify the number of working threads.
+
+* **Alter the Concurrency of Requests in the Code**: It's essential for businesses to iteratively test and evaluate the trade-off between the latency gains from increasing concurrency and the stability loss at peak levels to determine an optimal concurrency value.
+
+* **Use Batch Interfaces**: If the business scenario allows, replacing the current interface with a batch interface could be a more effective strategy.
+
+# Looking into the Future of Optimization
+
+## The Final Frontier: Kernel
+
+The only area currently open for further optimization is the Kernel.
```

Comment:
The translations is not exactly what it means in the original better.

The following is closer:

"Currently the only area we haven't explored for optimization is the Kernel."

The areas we discuss before may have room for further optimization and we're still working on some of them.

Reply:
Thanks!